### PR TITLE
Add OS name to _nodes and _cluster/nodes

### PR DESCRIPTION
**Pull request by bleskes**
we currently don't expose this.

This adds the following to the OS section of `_nodes`:

```
"os": {
     "name": "Mac OS X",
     ...
 }
```

 and the following to the OS section of `_cluster/stats`:

```
 "os": {
    ...
    "names": [
       {
          "name": "Mac OS X",
           "count": 1
       }
     ],
     ...
 },
```

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java [hunk 1]
```diff
@@ -339,6 +344,11 @@ public void readFrom(StreamInput in) throws IOException {
             availableProcessors = in.readVInt();
             availableMemory = in.readLong();
             int size = in.readVInt();
+            names = new ObjectIntHashMap<>(size);
+            for (; size > 0; size--) {
```
Comment (s1monw) [inline core/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java]: can we have a simple for loop here starting from 0?

File: core/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java [hunk 2]
```diff
@@ -303,10 +303,12 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
 
         int availableProcessors;
         long availableMemory;
+        ObjectIntHashMap<String> names;
```
Comment (s1monw) [inline core/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java]: can this be final? I mean do we need to initilaize lazy?

**Comment by s1monw** [comment]
left some comments

File: core/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java (hunk 2 above)
Comment (bleskes) [inline core/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java]: I think we can.

File: core/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java (hunk 1 above)
Comment (bleskes) [inline core/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java]: oh man, you insult the C++'er in me :p . will change.

**Comment by bleskes** [comment]
@s1monw pushed another commit

**Comment by s1monw** [comment]
LGTM

**Pull request closed without merging**